### ISSUE TITLE
Different optimizer

### ISSUE THREAD

**Issue by tommiekerssies** [MediaBrain-SJTU/RegAD#26]
When I change the optimizer, for example to adam, the loss goes to -1 after one or two epochs already. It seems to me the model is collapsing to a trivial solution. By using SGD with the low learning rate, this collapse is maybe just delayed. What are your thoughts on this?

**Comment by chaoqinhuang** [comment]
This is a famous undesired trivial solution to Siamese networks: all outputs “collapsing” to a constant. There have been several general strategies for preventing Siamese networks from collapsing. We follow the setting of the optimizer in Kaiming's [SimSiam](https://arxiv.org/pdf/2011.10566.pdf) (including SGD optimizer, cosine decay schedule for lr, weight decay, and the SGD momentum). We suggest not changing them in order to avoid the collapsing problem.
